### ISSUE TITLE
How to resume downloading from .part file

### ISSUE THREAD

**Issue by hasnogaems** [yt-dlp/yt-dlp#7669]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2023.07.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

I restarted my system on ubuntu and have .part file of unfinished download. The thing is I have no idea where url went. How can I resume my download without knowing url?

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
t1@t1-MEGABOOK-T1:~$ yt-dlp -v
[debug] Command-line config: ['-v']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2023.07.22.131037 [11de6fec9] (zip)
[debug] Python 3.10.6 (CPython x86_64 64bit) - Linux-5.19.9-0929-x86_64-with-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.11.0, brotli-1.0.9, certifi-2020.06.20, mutagen-1.45.1, pyxattr-0.7.2, secretstorage-3.3.1, sqlite3-2.6.0, websockets-9.1
[debug] Proxy map: {}
[debug] Loaded 1857 extractors
```

**Comment by bashonly** [comment]
You need the URL or at least an infojson (which you could use with `--load-info-json`)

Do you have the `id` in the filename? You probably do if you are using the default output template. You may be able to reconstruct the URL if you know the `id` and remember what site you were downloading from.

Otherwise you could check your shell history (press the Up arrow in your terminal) to see if the yt-dlp command was recorded

Without more information (filename, site, etc), there's not much anyone else can do to help

**Comment by SimonBaars** [comment]
In the folder with the `part` file, execute the following:
```
yt-dlp "[url]" --continue
```

**Comment by unitof** [comment]
If it was a livestream or a URL you cannot access anymore, `ffmpeg` can often repair truncated or unfinalized MP4 files:

```
ffmpeg -i aborted_input.mp4.part -c copy repaired_output.mp4
```

**Comment by SimonEast** [comment]
> In the folder with the `part` file, execute the following:
> 
> ```
> yt-dlp "[url]" --continue
> ```

`--continue` is supposedly on by default (according to the README), so why the need to specify it?

I did however just attempt to resume a stopped download by re-running the same initial command (without `--continue`) and it broke badly. Here's what happened:

1. I was downloading an HLS video (using the native downloader) when my internet cut out
2. yt-dlp retried the download 20 times, but it happened very fast (within a few seconds) since the internet was completely disconnected, and then it gave up and exited
3. I ran the same yt-dlp command that had started the download, and yt-dlp seemed to think that all video fragments were already downloaded (they were not) and proceeded to download the audio portion, mux them into a single file, and delete the separate audio and video streams.
4. I now have an MP4 file with only part of the video, but all of the audio. The rest is a black screen.

I'm not sure if there's any way to resume the 15gb download now that the file has been muxed. Perhaps I should have specified the `--keep-video` option. Would that have helped?

Anyway, surely this is not how it's supposed to work?
